app/js/components/orderEntry/OrdersTable.jsx

```diff
@@ -9,21 +9,86 @@ import DrugOrderDetails from './DrugOrderDetails';
 import fetchOrders from '../../actions/fetchOrders';
 import { DRUG_ORDER } from './orderTypes';
 import { setSelectedOrder } from '../../actions/orderAction';
+import createLabOrder from '../../actions/createLabOrder';
+import { successToast, errorToast } from '../../utils/toast';
+import toggleUrgency from '../../actions/labOrders/toggleUrgency';
 
 export class OrdersTable extends PureComponent {
+  state = {
+    recentUrgency: '',
+    previousOrderId: '',
+  };
+
   componentDidMount() {
     this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
   }
 
-  componentDidUpdate(prevProps, prevState) {
+  componentDidUpdate(prevProps) {
+    const { previousOrderId, recentUrgency } = this.state;
+    const {
+      status: { added, error },
+      errorMessage,
+      labOrderData,
+    } = this.props.createLabOrderReducer;
     if (this.props.patient.uuid !== prevProps.patient.uuid) {
       this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
     }
+    if (added && labOrderData !== prevProps.createLabOrderReducer.labOrderData) {
+      this.props.dispatch(toggleUrgency(
+        previousOrderId,
+        labOrderData.orders[0].uuid,
+        recentUrgency,
+      ));
+      successToast('urgency successfully changed');
+    }
+    if (error) {
+      errorToast(errorMessage);
+    }
   }
 
   handleActiveOrderEdit = (order) => {
-    this.props.dispatch(setSelectedOrder({ order: { ...order, status: 'EDIT' }, currentOrderType: DRUG_ORDER, activity: 'EDIT' }));
-  }
+    this.props.dispatch(setSelectedOrder({
+      order: { ...order, status: 'EDIT' },
+      currentOrderType: DRUG_ORDER,
+      activity: 'EDIT',
+    }));
+  };
+
+  toggleUrgency = (order) => {
```

Comment:
so looking at this file, there are different `togglUrgency`'s already. one is an action and the other is a handler for actually toggling urgency, do you mind renaming this to `handleUrgencyToggling` so it follows the convention of an event handler for clicks ?